src/include/migraphx/matcher.hpp

```diff
@@ -418,34 +418,41 @@ void find_matches_for(source_location location, Mod& mod, instruction_ref ins, M
                 return;
             if(trace > 1 and trace_for)
                 std::cout << "Match: " << matcher_name << std::endl;
-            auto elapsed_time = time<std::chrono::nanoseconds>([&] {
-                auto r = match_instruction(get_module(mod), ins, m.matcher());
-                if(r.result == get_module(mod).end())
+
+            timer match_timer;
+            auto r = match_instruction(get_module(mod), ins, m.matcher());
+            auto match_time = match_timer.record<std::chrono::nanoseconds>();
+
+            if(r.result == get_module(mod).end())
                 return;
-                if(trace > 0 or trace_for)
-                {
-                    std::cout << "Matched by " << matcher_name << std::endl;
-                    get_module(mod).debug_print(ins);
-                }
-                // If its already invalid dont validate it again
-                bool invalidated = validate and get_module(mod).validate() != get_module(mod).end();
-                m.apply(mod, r);
-                if(validate and not invalidated)
+            if(trace > 0 or trace_for)
+            {
+                std::cout << "Matched by " << matcher_name << std::endl;
+                get_module(mod).debug_print(ins);
+            }
+            // If its already invalid dont validate it again
+            bool invalidated = validate and get_module(mod).validate() != get_module(mod).end();
+                
+            timer apply_timer;
+            m.apply(mod, r);
+            auto apply_time = apply_timer.record<std::chrono::nanoseconds>();
```

Comment:
Use the `time` function.

Reply:
I have used the timer struct from /src/include/migraphx/time.hpp
Should I use separate `time` function for this?